src/formats/onenote.ts

```diff
@@ -142,150 +147,174 @@ export class OneNoteImporter extends FormatImporter {
-					if (index !== -1) {
-						this.selectedSections.splice(index, 1);
+			for (const section of entity.sections as OnenoteSection[]) {
+				const listElement = sectionList.createEl('li', {
+					cls: 'task-list-item',
+				});
+				let label = listElement.createEl('label');
+				let checkbox = label.createEl('input');
+				checkbox.type = 'checkbox';
+
+				label.appendChild(document.createTextNode(section.displayName!));
+				label.createEl('br');
+
+				checkbox.addEventListener('change', () => {
+					if (checkbox.checked) this.selectedIds.push(section.id!);
+					else {
+						const index = this.selectedIds.findIndex((sec) => sec === section.id);
+						if (index !== -1) {
+							this.selectedIds.splice(index, 1);
+						}
 					}
-				}
-			});
+				});
+
+				this.renderHierarchy(section, parentEl);				
+			}
 		}
 	}
 
 	async import(progress: ImportContext): Promise<void> {
-		// Remove possible duplicates, eg. when the user selects "Select all" with existing selections.
-		this.selectedSections = this.selectedSections.filter((item, index, array) => array.indexOf(item) === index);
-
-		let outputFolder = await this.getOutputFolder();
-		let remainingSections = this.selectedSections.length;
+		// Remove possible duplicate IDs, eg. when the user selects "Select all" with existing selections
+		this.selectedIds =  [...new Set(this.selectedIds)];
+		
+		let remainingSections = this.selectedIds.length;
 
-		if (!outputFolder) {
+		if (!await this.getOutputFolder()) {
 			new Notice('Please select a location to export to.');
 			return;
 		}
+		else this.outputFolder = (await this.getOutputFolder())!;
 
 		progress.status('Starting OneNote import');
 
-		for (let section of this.selectedSections) {
+		for (let sectionId of this.selectedIds) {
+			this.currentSection = sectionId;
 			progress.reportProgress(0, remainingSections);
 			remainingSections--;
-
 			let pageCount: number = 0;
 
-			let sectionFolder: TFolder;
-			if (section.parentSectionGroup) {
-				let sectionGroupFolder: TFolder = await this.createFolders(outputFolder.path + '/' + section.parentSectionGroup.displayName);
-				sectionFolder = await this.createFolders(sectionGroupFolder.path + '/' + section.displayName);
-			}
-			else sectionFolder = await this.createFolders(outputFolder.path + '/' + section.displayName);
+			const baseUrl = `https://graph.microsoft.com/v1.0/me/onenote/sections/${sectionId}/pages`;
+			const params = new URLSearchParams({
+			  $select: 'id,title,createdDateTime,lastModifiedDateTime,level,order,contentUrl',
+			  $orderby: 'order',
+			  pagelevel: 'true'
+			});
 
-			const pagesUrl = `https://graph.microsoft.com/v1.0/me/onenote/sections/${section.id}/pages?$select=id,title,createdDateTime,lastModifiedDateTime`;
-			let pages: OnenotePage[] = (await this.fetchResource(pagesUrl, 'json')).value;
+			const pagesUrl = `${baseUrl}?${params.toString()}`;
+	
+			let pages: OnenotePage[] = ((await this.fetchResource(pagesUrl, 'json')).value);
```

Comment:
Can we put these pages on the section when we retrieve them rather than doing it in `getEntityPath`?